[8.0] skip invoice lines with value 0

luc-demeyer opened the pull request.

luc-demeyer committed:
[8.0] skip invoice lines with value 0

cubells reviewed (approved):
:+1: 

LGTM

Code review and test.

luc-demeyer committed:
Update intrastat_product_declaration.py

@cubells
I changed the check on 'price_unit' with value into price_subtotal.
By doing so we also cover the case of invoice lines with 100% discount.
Let me know if it's ok so that we can merge (and forward port to v10)

luc-demeyer commented:
@cubells
I changed the check on 'price_unit' with value 0.0 into 'price_subtotal'.
By doing so we also cover the case of invoice lines with 100% discount.
Let me know if it's ok so that we can merge (and forward port to v10)

cubells commented on intrastat_product/models/intrastat_product_declaration.py:
```diff
@@ -501,6 +501,9 @@ def _gather_invoices(self):
 
+                if not inv_line.price_subtotal:
+                    continue
+                
```
Remove all the white spaces of this line to avoid lint errors.

pedrobaeza replied:
A better expression is:
```python
for inv_line in invoice.invoice_line.filtered('price_subtotal'):
    ...
```

cubells reviewed (approved):
@luc-demeyer 

If you remove white spaces all is ok for me.

pedrobaeza commented:
Travis is red

luc-demeyer committed:
fix flake8
